test/CMakeLists.txt

```diff
@@ -8,25 +8,61 @@ set(CMAKE_C_FLAGS "-fprofile-arcs -ftest-coverage")
 set(CMAKE_EXE_LINKER_FLAGS "-fprofile-arcs -ftest-coverage -coverage")
 
 set(repo_root ..)
+enable_testing()
+
+# Build Unity
+
 add_library(unity STATIC ${repo_root}/external/unity/src/unity.c)
 target_compile_definitions(unity PUBLIC -DUNITY_INCLUDE_CONFIG_H)
 target_include_directories(unity PUBLIC ${repo_root}/external/unity/src .)
 
+# Build FFF
+
 add_library(fff INTERFACE)
 target_include_directories(fff INTERFACE ${repo_root}/external/fff)
 
-enable_testing()
+# Build cJSON
+
+set(cjson_dir "${repo_root}/external/cJSON")
+add_library(cjson
+    "${cjson_dir}/cJSON.c"
+    "${cjson_dir}/cJSON_Utils.c"
+)
+target_include_directories(cjson PUBLIC "${cjson_dir}")
+
+# Function for declaring unit tests
+
+function(golioth_unit_test name src)
+    add_executable(${name} ${src} ${ARGN})
+    target_include_directories(${name} PRIVATE
+        ${repo_root}/src/priv_include
+        ${repo_root}/src/include
+    )
+    target_link_libraries(${name} unity fff)
+    add_test(NAME ${name} COMMAND "./${name}")
+endfunction()
 
 #
 # Test executables
 #
-set(test_name "test_ringbuf")
-add_executable(${test_name}
+
+# Ringbuf unit tests
+
+golioth_unit_test(test_ringbuf
     ${repo_root}/src/golioth_ringbuf.c
     test_ringbuf.c
 )
-target_include_directories(${test_name} PRIVATE
-    ${repo_root}/src/priv_include
+
+# RPC unit tests
+
+golioth_unit_test(test_golioth_rpc
+    test_golioth_rpc.c
+    fakes/golioth_coap_client_fake.c
+    fakes/golioth_time_fake.c
+)
+target_include_directories(test_golioth_rpc PRIVATE
+    ${repo_root}/external/libcoap/include
+    ${repo_root}/port/esp_idf/libcoap/include
+    ${repo_root}/port/linux
```

Comment:
Why do we include both `esp_idf` as well as `linux` port include directories? Shouldn't single port directory be included?